lib/jekyll/filters.rb

```diff
@@ -373,8 +373,8 @@ def time(input)
       when Numeric
         Time.at(input)
       else
-        Jekyll.logger.error "Invalid Date:", "'#{input}' is not a valid datetime."
-        exit(1)
+        raise Errors::InvalidDateError,
+          "Invalid Date: '#{input}' is not a valid datetime."
```

Comment:
@ayastreb When an error is raised in Liquid, the renderer catches that and prints the document for which the error occurred so that's all take care of. What's important is to ensure the input is displayed properly. How about `"Invalid Date: \"#{input.inspect}\" is not a valid time."` ?